src/mlir/generator.cc

```diff
@@ -551,21 +585,28 @@ namespace mlir::verona
     auto name = AST::getID(ast);
 
     // Get both arguments
-    // TODO: If the arguments are tuples, do we need to apply element-wise?
     auto arg0 = parseNode(AST::getOperand(ast, 0).lock());
     if (auto err = arg0.takeError())
       return std::move(err);
+    auto lhs = arg0->get();
     auto arg1 = parseNode(AST::getOperand(ast, 1).lock());
     if (auto err = arg1.takeError())
       return std::move(err);
+    auto rhs = arg1->get();
+    // We don't know what the final type will be so we assume it's either of the
+    // input types, for now, if they're different.
+    // If the functions are known (ex. comparisons) we set the known retTy.
```

Comment:
Again, I would sooner we not try to do type checking/inference at this point.  We should do that in a single pass later. If you don't know the type use unknown.

Reply:
There are two problems this avoid (not solves):
1. Unknown types are invalid in the type checker for anything. Sub-types, viewpoint adaptation, etc. The more unknowns I have the less I can do.
2. If I continue to propagate unknowns as if they were "poison", then eventually I'd only have unknown, and would not be able to find what class type a value has to be able to extract its field information.

The first is probably fixed by improving the type checking. The latter would have to follow the type and all the paths across the CFG to "know" (or best guess) what the class type is.

Reply:
"best guess" should be "unknown"